dhall/src/Dhall/Normalize.hs

```diff
@@ -920,6 +920,14 @@ normalizeWithM ctx e0 = loop (Syntax.denote e0)
                                     Nothing -> Merge x' y' <$> t'
                             _ ->
                                 Merge x' y' <$> t'
+                    Some a ->
+                        case Dhall.Map.lookup "Some" kvsX of
+                            Just vX -> loop (App vX a)
+                            Nothing -> Merge x' y' <$> t'
```

Comment:
I was a bit surprised of this pattern of simply ignoring ill-typed code. Then again I'm not sure who's using `normalizeWithM` at all?!

Reply:
@sjakobi: It's been that way since the very beginning